app/feature/feature-insurances/src/main/kotlin/com/hedvig/android/feature/insurances/insurance/InsuranceDestination.kt

```diff
@@ -322,6 +334,85 @@ private fun CrossSellItem(crossSell: CrossSell, onCrossSellClick: (String) -> Un
   }
 }
 
+@Composable
+private fun InsuranceScreenWithPlaceholders() {
+  Column {
+    Spacer(Modifier.height(16.dp))
+    InsuranceCardPlaceHolder(
+      modifier = Modifier
+        .padding(horizontal = 16.dp)
+        .clip(MaterialTheme.shapes.squircleMedium),
+    )
+    Spacer(Modifier.height(32.dp))
+    NotificationSubheading(
+      text = stringResource(R.string.insurance_tab_cross_sells_title),
+      showNotification = false,
+      modifier = Modifier.padding(horizontal = 16.dp),
+    )
+    Spacer(Modifier.height(16.dp))
+    HorizontalDivider(Modifier.padding(horizontal = 16.dp))
+    Spacer(Modifier.height(16.dp))
+    List(5) {
+      CrossSellItemPlaceHolder(
+        modifier = Modifier.padding(horizontal = 16.dp),
+      )
+    }
+  }
+}
+
+@Composable
+private fun CrossSellItemPlaceHolder(modifier: Modifier = Modifier) {
```

Comment:
the placeholder does not strictly follow the shape of Composable, it adds a bit of overlap. That's why if we add two lines of text for placeholder it looks a bit ugly; and it still moves around a bit when it loaded, bc some crossCells have 1 row of text, and some two.

Reply:
I think the same logic applies here as what I describe in the profile screen.
For these placeholders to still be able to follow the changes of the real UI if they ever come, they need to share more of their UI code.
If that is by having a common parent composable, or by both states being actually the exact same composable but with having `placeholder(visible = {state is Loading})` is probably something that should be decided on a case by case basis.
But if they are just copy pasted UI it is bound to have slight differences at least down the road.